### PR TITLE
Test for activity description

### PR DESCRIPTION
**Pull request by RoxaneBurri**

### CODE REVIEW

File: app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityTest.java [hunk 1]
```diff
@@ -90,31 +106,53 @@ public void onComplete(@NonNull Task<AuthResult> task) {
...
+
+        try {
+            Thread.sleep(1500);
+        } catch (InterruptedException e) {
+            assert (false);
+        }
 
-    /*@Before
-    public void setUp(){
-        fStore = mock(FirebaseFirestore.class);
```
Comment (kotleta2007) [inline app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityTest.java:98]: It's a good idea to remove commented out code if it's not used

File: app/src/androidTest/java/com/sdp/movemeet/view/activity/ListActivityTest.java [hunk 2]
```diff
@@ -17,38 +17,11 @@
...
-        fAuth.signInWithEmailAndPassword("movemeet@gmail.com", "password").addOnSuccessListener(new OnSuccessListener<AuthResult>() {
-            @Override
-            public void onSuccess(AuthResult authResult) {
-                latch.countDown();
-            }
-        }).addOnFailureListener(new OnFailureListener() {
-            @Override
-            public void onFailure(@NonNull Exception e) {
-                assert (false);
-            }
```
Comment (kotleta2007) [inline app/src/androidTest/java/com/sdp/movemeet/view/activity/ListActivityTest.java:40]: Why did you remove this test? Is this code covered by some other test now?

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -138,7 +138,7 @@
     <string name="activity_description_date">date</string>
     <string name="activity_description_duration">Duration</string>
     <string name="activity_description_address">Address</string>
-    <string name="activity_description_description">Address</string>
```
Comment (kotleta2007) [inline app/src/main/res/values/strings.xml:141]: Thanks for fixing this, it didn't make much sense when I saw it in the app

**Review by kotleta2007 — APPROVED** [review]

File: app/src/androidTest/java/com/sdp/movemeet/view/activity/ListActivityTest.java (hunk 2 above)
Comment (RoxaneBurri) [reply app/src/androidTest/java/com/sdp/movemeet/view/activity/ListActivityTest.java:40]: I empty the ActivityListActivity for the moment, because this activity do nothing and lower the test coverage

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (RoxaneBurri) [reply app/src/main/res/values/strings.xml:141]: Yes !

File: app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityTest.java [hunk 4]
```diff
@@ -90,31 +106,53 @@ public void onComplete(@NonNull Task<AuthResult> task) {
...
         Intents.release();
     }
 
-    /*@Test
-    public void createTitleViewTest(){
-        ViewGroup view = inflate(R.layout.activity_description);
-        TextView textView = view.findViewById(R.id.activity_title_description);
+    @Test
+    public void chatButtonIsCorrect() {
+        Intents.init();
```
Comment (Walfar) [inline app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityTest.java:141]: Couldn't the init for Intents be refactored in a @Before, since it is used in many tests ? And the release could be done in the @After

File: app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityTest.java [hunk 5]
```diff
@@ -90,31 +106,53 @@ public void onComplete(@NonNull Task<AuthResult> task) {
...
+        //onView(withId(R.id.activity_date_description)).check(matches(withText(String.valueOf(DUMMY_DATE))));
+        onView(withId(R.id.activity_address_description)).check(matches(withText(DUMMY_ADDRESS)));
+        onView(withId(R.id.activity_sport_description)).check(matches(withText(String.valueOf(DUMMY_SPORT))));
+        //onView(withId(R.id.activity_duration_description)).check(matches(withText(String.valueOf(DUMMY_DURATION))));
+
+        onView(withId(R.id.activity_organisator_description)).check(matches(withText(DUMMY_ORGANISATOR_ID)));
+        //onView(withId(R.id.activity_number_description)).check(matches(withText(String.valueOf(DUMMY_NUMBER_PARTICIPANT))));
+        //onView(withId(R.id.activity_participants_description)).check(matches(withText(String.valueOf(DUMMY_PARTICIPANTS_ID.size()))));
+
+        try {
```
Comment (Walfar) [inline app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityTest.java:124]: Is this sleep necessary ?

**Review by Walfar — APPROVED** [review]

File: app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityTest.java (hunk 4 above)
Comment (RoxaneBurri) [reply app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityTest.java:141]: It's true, I will do it in my next pull request

File: app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityTest.java (hunk 5 above)
Comment (RoxaneBurri) [reply app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityTest.java:124]: Nop

Comment (RoxaneBurri) [reply app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityTest.java:124]: Nop

File: app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityUnregisterTest.java [hunk 6]
```diff
@@ -55,15 +76,39 @@
...
+    @Test
+    public void loginButtonIsCorrect(){
+        Intents.init();
+        try {
+            Thread.sleep(2000);
+        } catch (InterruptedException e) {
+            assert (false);
+        }
+        onView(withId(R.id.activityGoDescription)).perform(scrollTo(), click());
+        intended(hasComponent(LoginActivity.class.getName()));
```
Comment (Antho1426) [inline app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityUnregisterTest.java:111]: What does this line check actually? What does "LoginActivity.class.getName()" return?

File: app/src/androidTest/java/com/sdp/movemeet/view/activity/ListActivityTest.java (hunk 2 above)
Comment (kotleta2007) [reply app/src/androidTest/java/com/sdp/movemeet/view/activity/ListActivityTest.java:40]: Ok, that makes sense to me

**Review by Antho1426 — APPROVED** [review]
